### PR TITLE
add nightly ci

### PR DESCRIPTION
**Pull request by ddengster**
Signed-off-by: ddengster <ed.fan@osrfoundation.org>

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -1,5 +1,10 @@
 name: build
-on: [push, pull_request]
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 0 * * *'
```
Comment (mxgrey) [inline .github/workflows/build.yaml]: Not that it matters too much, but [GitHub says](https://docs.github.com/en/actions/reference/events-that-trigger-workflows)

> Note: The schedule event can be delayed during periods of high loads of GitHub Actions workflow runs. High load times include the start of every hour. To decrease the chance of delay, schedule your workflow to run at a different time of the hour.

Should we be courteous and pick a somewhat randomized hour and minute for these periodic runs?

Comment (ddengster) [reply .github/workflows/build.yaml]: well i'll change to a random number but we really need see a graph of github's traffic load to see the effectiveness

Comment (ddengster) [reply .github/workflows/build.yaml]: 23bc15d

Comment (mxgrey) [reply .github/workflows/build.yaml]: Yeah, I doubt we'll notice any difference, but I figure if they find it significant enough to make a note about it, then we may as well be courteous users and randomize :shrug:

**Review by mxgrey — APPROVED** [review]

**Pull request merged**